bitsandbytes/cuda_setup/main.py

```diff
@@ -111,7 +118,9 @@ def manual_override(self):
                           'For example by adding the following to your .bashrc: export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:<path_to_cuda_dir/lib64\n'
                           f'Loading CUDA version: BNB_CUDA_VERSION={os.environ["BNB_CUDA_VERSION"]}'
                           f'\n{"="*80}\n\n'))
-                    self.binary_name = self.binary_name[:-6] + f'{os.environ["BNB_CUDA_VERSION"]}.so'
+                    binary_name = self.binary_name.rsplit(".", 1)[0]
```

Comment:
This is a bit of a mess (based on my messy previous code). I have not looked at the details. Is there a way where we can clean this up slightly?